docs-src/security-model.md

```diff
@@ -0,0 +1,148 @@
+  GatewayClass.
+
+## Roles
+For the purposes of this security model, 3 common roles have been identified:
+
+* **Infrastructure provider**: The infrastructure provider (infra) is
+  responsible for the overall environment that the cluster(s) are operating in.
+  Examples include: the cloud provider (AWS, Azure, GCP, ...), the PaaS provider
+  in a company.
+* **Cluster operator**: The cluster operator (ops) is responsible for
+  administration of entire clusters. They manage policies, network access,
+  application permissions.
+* **Application developer**: The application developer (dev) is responsible for
+  defining their application configuration (e.g. timeouts, request
+  matching/filter) and Service composition (e.g. path routing to backends). 
+
+Although these roles can cover a wide variety of use cases, some organizations
+may be structured slightly differently. Many organizations may also have a
+fourth role that sits between "cluster operator" and "application developer":
+
+* **Application admin**: The application admin has administrative access to some
+  namespaces within a cluster, but not the cluster as a whole.
+
+## The Security Model
+There are two primary components to the Service APIs security model: RBAC and
+namespace restrictions.
+
+## RBAC
+RBAC (role-based access control) is the standard used for Kubernetes
+authorization. This allows users to configure who can perform actions on
+resources in specific scopes. RBAC can be used to enable each of the roles
+defined above. In most cases, it will be desirable to have all resources be
+readable by most roles, so instead we'll focus on write access for this model.
+
+### Write Permissions for Simple 3 Tier Model 
+| | GatewayClass | Gateway | Route | TrafficSplit |
+|-|-|-|-|-|
+| Infrastructure Provider | Yes | Yes | Yes | Yes |
+| Cluster Operators | No | Yes | Yes | Yes |
+| Application Developers | No | No | Yes | Yes |
+
+### Write Permissions for Advanced 4 Tier Model 
+| | GatewayClass | Gateway | Route | TrafficSplit |
+|-|-|-|-|-|
+| Infrastructure Provider | Yes | Yes | Yes | Yes |
+| Cluster Operators | Sometimes | Yes | Yes | Yes |
+| Application Admins | No | In Specified Namespaces | In Specified Namespaces | In Specified Namespaces |
+| Application Developers | No | No | In Specified Namespaces | In Specified Namespaces |
+
+## Namespace Restrictions
+The extra configuration options are not possible to control with RBAC. Instead,
+they will be controlled with configuration fields on GatewayClasses:
+
+* **gatewayNamespaceSelector**: This field is a selector of valid namespaces
+  that Gateways can use this GatewayClass from. An empty selector (default)
+  indicates that Gateways can use this GatewayClass from any namespace.
+
+* **routeNamespaceSelector**: This field is a selector of valid namespaces that
+  gateways can reference routes from. An empty selector (default) indicates that
+  routes can be referenced from any namespace.
```

Comment:
My current understanding is that this is in addition to the selectors on the Gateway resource. @danehans's PR implements the Gateway-level route selector here: https://github.com/kubernetes-sigs/service-apis/pull/12/files#diff-e7d01144441bba257ce9bedd33a139c6R241.

Is that understand correct?

Also, should we add the Gateway-level selectors to this security model?
I've a feeling that we may not have consensus yet and that's why you didn't put those in this PR.

Reply:
Yeah this is confusing, this was added with that in mind, but I probably need to find a better way to communicate the intent. This is intended to _limit_ the namespaces routes can be referenced in. This is entirely intended to be an authorization based attribute. The route namespace and labels selectors proposed by @danehans on Gateways would be in addition to this. 

As an example, if the GatewayClass only allowed reference Routes in namespaces A and B, and the Gateway referenced Routes in namespaces A, B, and C, the controller should only implement the Routes in namespaces A and B.

The whole concept is based around the concept that in some cases admins may want to allow Gateway creation in certain namespaces _and_ limit what routes those lower security Gateways can reference. IE, let any developer create Gateways with the `InternalDevGateway` class, but those can only reference Routes in namespaces with a `env: dev` label.

Reply:
I've updated the docs around this now, hopefully this is a bit clearer now.

Reply:
The reword makes it clearer. Thank you, Rob.